### ISSUE TITLE
The sidebar compass indicator does not goes red for offscreen monster that can see me

### ISSUE THREAD

**Issue by gettingusedto** [CleverRaven/Cataclysm-DDA#66801]
### Describe the bug

For whatever reason, the sidebar compass indicator does not goes red to indicate the offscreen monsters that sees the player despite being the case in past builds.

### Attach save file

N/A

### Steps to reproduce

1. get a sidebar that has compass, in my case i used the Compact sidebar
2. spawn a monster
3. walk away till it's offscreen
4. look at the compass

### Expected behavior

the compass indicator goes red for offscreen monsters that sees the player, like in past builds.

### Versions and configuration

- OS: Android
    - OS Version: Manufacturer: samsung; Model: SM-A710F; Release: 7.0; Incremental: A710FXXU2CRC2; 
- Game Version: 6585b7f [32-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Mind Over Matter [mindovermatter],
    Xedra Evolved [xedra_evolved],
    Bombastic Perks [bombastic_perks]
]

**Comment by Inglonias** [comment]
If you've set custom safe mode rules, I believe that changes the compass' behavior such that it stops doing this. Do you have custom safe mode rules in place?

**Comment by gettingusedto** [comment]
After checking the safe mode rule, i realized that i forgot it had a custom safe mode rule from something i was testing. Man, that wasnt obvious that the custom safe mode rule did it at all. deleting the one custom safemode rule does fix the issue. Sorry for the inconvenience.

**Comment by Inglonias** [comment]
Yeah. The interaction between safe mode rules and the compass are weird and not explained anywhere. The only reason I know about it at all is because of the work done for the above PR.
